# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: third_party/rust/wgpu-hal/src/gles/fence.rs
# Commit: b3109ad14685
# Full Hash: b3109ad14685624ada7861083ff5332aec9a7818
# Author: Andy Leiserson <aleiserson@mozilla.com>
# Date: 2025-10-07 17:00 +0000
# Description:
#   Bug 1992842 - Update wgpu to upstream 1072b878 (2025-10-06) r=webgpu-reviewers,supply-chain-reviewers,ErichDonGubler
# ==============================================================================

diff -r 1d3f000cf1f9 -r b3109ad14685 third_party/rust/wgpu-hal/src/gles/fence.rs
--- a/third_party/rust/wgpu-hal/src/gles/fence.rs	Tue Oct 07 16:51:44 2025 +0000
+++ b/third_party/rust/wgpu-hal/src/gles/fence.rs	Tue Oct 07 17:00:40 2025 +0000
@@ -102,7 +102,7 @@
         &self,
         gl: &glow::Context,
         wait_value: crate::FenceValue,
-        timeout_ns: u64,
+        timeout_ns: u32,
     ) -> Result<bool, crate::DeviceError> {
         let last_completed = self.last_completed.load(Ordering::Acquire);
 
@@ -134,7 +134,7 @@
             gl.client_wait_sync(
                 gl_fence.sync,
                 glow::SYNC_FLUSH_COMMANDS_BIT,
-                timeout_ns as i32,
+                timeout_ns.min(i32::MAX as u32) as i32,
             )
         };
 